File: Pertemuan 12/App Client with Login/Petugas.php
<?php
//Simpanlah dengan nama file : Petugas.php
require_once 'database.php';
class Petugas 
{
    private $db;
    private $table = 'petugas';
    public $kode_petugas = "";
    public $nama_petugas = "";
    public $jabatan_petugas = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_kode_petugas(int $kode_petugas)
    {
        $query = "SELECT * FROM $this->table WHERE kode_petugas = $kode_petugas";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`kode_petugas`,`nama_petugas`,`jabatan_petugas`) VALUES ('$this->kode_petugas','$this->nama_petugas','$this->jabatan_petugas')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET kode_petugas = '$this->kode_petugas', nama_petugas = '$this->nama_petugas', jabatan_petugas = '$this->jabatan_petugas' 
        WHERE id_petugas = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_kode_petugas($kode_petugas): int
    {
        $query = "UPDATE $this->table SET kode_petugas = '$this->kode_petugas', nama_petugas = '$this->nama_petugas', jabatan_petugas = '$this->jabatan_petugas' 
        WHERE kode_petugas = $kode_petugas";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id_petugas = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_kode_petugas($kode_petugas): int
    {
        $query = "DELETE FROM $this->table WHERE kode_petugas = $kode_petugas";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>

File: Pertemuan 12/App Client with Login/Petugas.py
import requests
import json
class Petugas:
    def __init__(self):
        self.__id=None
        self.__kode_petugas = None
        self.__nama_petugas = None
        self.__jabatan_petugas = None
        self.__url = "http://localhost/Myperpustakaan/petugas_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def kode_petugas(self):
        return self.__kode_petugas
        
    @kode_petugas.setter
    def kode_petugas(self, value):
        self.__kode_petugas = value
    @property
    def nama_petugas(self):
        return self.__nama_petugas
        
    @nama_petugas.setter
    def nama_petugas(self, value):
        self.__nama_petugas = value
    @property
    def jabatan_petugas(self):
        return self.__jabatan_petugas
        
    @jabatan_petugas.setter
    def jabatan_petugas(self, value):
        self.__jabatan_petugas = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_kode_petugas(self, kode_petugas):
        url = self.__url+"?kode_petugas="+kode_petugas
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id_petugas']
            self.__kode_petugas = item['kode_petugas']
            self.__nama_petugas = item['nama_petugas']
            self.__jabatan_petugas = item['jabatan_petugas']
        return data
    def simpan(self):
        payload = {
            "kode_petugas":self.__kode_petugas,
            "nama_petugas":self.__nama_petugas,
            "jabatan_petugas":self.__jabatan_petugas
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_kode_petugas(self, kode_petugas):
        url = self.__url+"?kode_petugas="+kode_petugas
        payload = {
            "kode_petugas":self.__kode_petugas,
            "nama_petugas":self.__nama_petugas,
            "jabatan_petugas":self.__jabatan_petugas
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_kode_petugas(self,kode_petugas):
        url = self.__url+"?kode_petugas="+kode_petugas
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text
